src/extractor/guidance/roundabout_handler.cpp

```diff
@@ -478,13 +478,18 @@ Intersection RoundaboutHandler::handleRoundabouts(const RoundaboutType roundabou
                                 return true;
                         }
                         return false;
-                    }();
+                    };
 
-                    if (has_non_ignorable_exit)
+                    if (out_data.road_classification.IsLowPriorityRoadClass() ||
+                        has_non_ignorable_exit())
```

Comment:
I guess this check could benefit from some additional comment `//count normal exits and service roads, if the roundabout is a service road itself`

Other than this, this looks great.